artifactory/utils/buildinfoproperties_test.go

```diff
@@ -117,9 +143,32 @@ func compareViperConfigs(t *testing.T, actual, expected *viper.Viper, projectTyp
 	}
 }
 
-func TestSetProxyIfNeeded(t *testing.T) {
-	proxyOrg := getOriginalProxyValue()
-	setProxy(proxy, t)
+func TestSetHttpProxy(t *testing.T) {
+	backupProxyPass := os.Getenv(httpProxy + Password)
+	os.Setenv(httpProxy+Password, "")
+	setAndAssertProxy(httpProxyForTest, t)
+	vConfig := viper.New()
+
+	err := setProxyIfDefined(vConfig)
+	if err != nil {
+		t.Error(err)
```

Comment:
Let's try to avoid using `t.Error`. We can flatten the code by using, for example `assert.NoError(t, err)` in this line.
I suggest taking this opportunity and replacing them in all locations in this file.